ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/slot/SlotStore.java

```diff
@@ -65,15 +65,21 @@ public SlotStore(SlotStoreEnvironmentBean config) throws IOException {
 
         // unused for now, but eventually we'll have a disk backend...
         File storeDir = new File(config.getStoreDir());
-        if (!storeDir.exists() && !storeDir.mkdirs()) {
-            throw new IOException(tsLogger.i18NLogger.get_dir_create_failed(storeDir.getCanonicalPath()));
-        }
         storeDirCanonicalPath = storeDir.getCanonicalPath();
 
-        slots = new RAMSlots(config);
+        slots = config.getBackingSlots();
+        slots.init(config);
 
+        // internal recovery to rebuild the slotIdIndex and freeList
```

Comment:
internal recovery in the case of config changes is complex and the current situation is basically 'don't try it'. The extent to which we want to support and test it is open to discussion. Broadly I don't think it will be common to change the config on a non-empty store, except to grow the number of slots or the slotSize, both of which will work with the current implementation. Relocating the directory should likewise work - it will change the store name, but that's not used anywhere except for debugging output.

Reply:
So can you add a note to the config bean to say "don't do XYZ" if the store is not empty.

Reply:
Once this conversation is resolved we can merge the contribution.

Reply:
Warnings added to javadoc.